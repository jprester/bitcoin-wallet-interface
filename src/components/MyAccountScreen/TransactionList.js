import React from 'react';
import { uniqueId } from 'lodash';
import PropTypes from 'prop-types';

import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemSecondaryAction from '@material-ui/core/ListItemSecondaryAction';
import ListItemText from '@material-ui/core/ListItemText';

const TransactionList = ({ transactionList, classes }) => {
  if (!transactionList.length) {
    return (
      <p> List is empty </p>
    );
  }

  return (transactionList.length &&
  <List>
    {transactionList.map(transaction => (
      <ListItem key={uniqueId('list-item')} dense disableGutters divider className={classes.listItem}>
        <ListItemText className={classes.transactionAdress} primary={transaction.address} />
        <ListItemSecondaryAction className={classes.transactionAmount}>
          {transaction.amount} BTC
        </ListItemSecondaryAction>
      </ListItem>
    ))}
  </List>
  );
};

export default TransactionList;


TransactionList.propTypes = {
  transactionList: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
  classes: PropTypes.shape({}).isRequired,
};

