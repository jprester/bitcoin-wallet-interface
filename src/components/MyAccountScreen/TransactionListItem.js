import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';

const styles = {
  root: {
    width: '100%',
  },
};

const date = `${new Date().getDate()}.${new Date().getMonth()}.${new Date().getFullYear()} `;

console.log(date);

const TransactionListItem = props => (
  <li>
    <p className="list-item-info">
      <span className="list-item-info-date">{date}</span>
      <span className="list-item-info-arrow" />
      <span className="list-item-info-amount">{props.amount}</span>
    </p>
    <p className="list-item-address">{props.address}</p>
  </li>
);

export default withStyles(styles)(TransactionListItem);

TransactionListItem.propTypes = {
  amount: PropTypes.number,
  address: PropTypes.string,
};

TransactionListItem.defaultProps = {
  amount: 0,
  address: '',
};
