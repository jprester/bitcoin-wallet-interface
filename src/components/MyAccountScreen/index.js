import React from 'react';

import TransactionsListWidgetMain from './TransactionsListWidget';

export default function MyAccountScreen() {
  return (
    <div className="content-block">
      <TransactionsListWidgetMain />
    </div>
  );
}
