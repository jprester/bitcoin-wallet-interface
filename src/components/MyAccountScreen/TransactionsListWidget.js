import React from 'react';
import { connect } from 'react-redux';
import { withStyles } from '@material-ui/core/styles';
import PropTypes from 'prop-types';

import TransactionList from './TransactionList';


const styles = theme => ({
  root: {
    width: '100%',
    maxWidth: 360,
    marginTop: 50,
    backgroundColor: theme.palette.background.paper,
  },

  transactionAdress: {
    fontSize: 11,
  },

  transactionAmount: {
    fontSize: 11,
  },
});

export const TransactionsListWidget = (props, classes) => (
  <div className={classes.root}>
    <h2> Transactions </h2>
    <TransactionList classes={classes} {...props} />
  </div>
);

const mapStateToProps = state => ({
  transactionList: state.transactionsData.transactions,
});

const styledComponent = withStyles(styles)(TransactionsListWidget);

export default connect(mapStateToProps)(styledComponent);

TransactionsListWidget.propTypes = {
  classes: PropTypes.shape({}).isRequired,
};
