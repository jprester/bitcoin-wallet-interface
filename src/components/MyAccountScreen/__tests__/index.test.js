import React from 'react';
import { shallow } from 'enzyme';
import MyAccountScreen from '../index';

const wrapper = shallow((<MyAccountScreen />));

describe('index', () => {
  it('should render MyAccountScreen correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
