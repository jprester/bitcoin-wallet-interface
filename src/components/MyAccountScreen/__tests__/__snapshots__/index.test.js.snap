// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index should render MyAccountScreen correctly 1`] = `
<div
  className="content-block block-size-1-2"
>
  <Connect(WithStyles(Component)) />
</div>
`;
